include/deal.II/multigrid/mg_smoother.h

```diff
@@ -642,7 +648,16 @@ namespace mg
     if (this->debug > 0)
       deallog << 'S' << level << ' ';
 
-    for (unsigned int i=0; i<steps2; ++i)
+    if (set_u_to_zero)
+      {
+        if (T)
+          (*this)[level].Tvmult(u, rhs);
+        else
+          (*this)[level].vmult(u, rhs);
+        if (this->symmetric)
+          T = !T;
+      }
+    for (unsigned int i=set_u_to_zero; i<steps2; ++i)
```

Comment:
I think this cast from bool to int is weird. Can you be more specific by doing something like ``(set_u_to_zero)?1:0``?